front/src/monthview/Monthview.js

```diff
@@ -0,0 +1,367 @@
+    const link = moment(`${this.state.yearNro}-${this.state.monthNro}-1`, 'YYYY-MM-DD');
+
+    if (this.state.yearNro === 0) {
+      return;
+    }
+
+    while (apuAlku.isoWeek() !== endWeek) {
+      table.push(
+        <Link class="link" to={`/weekview/${link.format('YYYY-MM-DD')}`}>
+          <div>
+            {startWeek}
+          </div>
+        </Link>,
+      );
+      link.add(1, 'weeks');
+      apuAlku.add(1, 'weeks');
+      startWeek = apuAlku.isoWeek();
+    }
+
+    table.push(
+      <Link class="link" to={`/weekview/${link.format('YYYY-MM-DD')}`}>
+        <div>
+          {startWeek}
+        </div>
+      </Link>,
+    );
+
+    return table;
+  }
+
+  previousMonthClick(e) {
+    e.preventDefault();
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramYear = urlParamDateSplit[0];
+      const paramMonth = urlParamDateSplit[1].padStart(2, '0');
+      const paramDay = urlParamDateSplit[2];
+
+      const paramDateCorrect = moment(paramYear + paramMonth + paramDay, 'YYYYMMDD');
+      paramDateCorrect.add(1, 'days');
+      paramDateCorrect.subtract(1, 'month');
+      this.props.history.replace(`/Monthview/${paramDateCorrect.toISOString().substring(0, 10)}`); // eslint-disable-line
+      this.setState({
+        monthNro: paramDateCorrect.format('MM'),
+        yearNro: paramDateCorrect.format('YYYY'),
+      });
+    } catch (err) {
+      this.update();
+      console.error(err);
+    }
+  }
+
+  nextMonthClick(e) {
+    e.preventDefault();
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramYear = urlParamDateSplit[0];
+      const paramMonth = urlParamDateSplit[1].padStart(2, '0');
+      const paramDay = urlParamDateSplit[2];
+
+      const paramDateCorrect = moment(paramYear + paramMonth + paramDay, 'YYYYMMDD');
+      paramDateCorrect.add(1, 'days');
+      paramDateCorrect.add(1, 'month');
+      this.props.history.replace(`/Monthview/${paramDateCorrect.toISOString().substring(0, 10)}`); // eslint-disable-line
+      this.setState({
+        monthNro: paramDateCorrect.format('MM'),
+        yearNro: paramDateCorrect.format('YYYY'),
+      });
+    } catch (err) {
+      this.update();
+      console.error(err);
+    }
+  }
+
+  update() {
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramYear = urlParamDateSplit[0];
+      const paramMonth = urlParamDateSplit[1].padStart(2, '0');
+
+      if((paramMonth !== this.state.monthNro && this.state.monthNro !== 0) || paramYear !== 0 && paramYear !== this.state.yearNro){ // eslint-disable-line
```

Comment:
no need to disable elsint

Reply:
Lint gives "Unexpected mix of '||' and '&&'" errors if removed

Reply:
Wrap them in brackets so it's in (A & B) || (C & D)

```suggestion
      if ((paramMonth !== this.state.monthNro && this.state.monthNro !== 0)
        || (paramYear !== 0 && paramYear !== this.state.yearNro)) {
```